homeassistant/components/mazda/sensor.py

```diff
@@ -0,0 +1,264 @@
+"""Platform for Mazda sensor integration."""
+from homeassistant.const import (
+    CONF_UNIT_SYSTEM_IMPERIAL,
+    LENGTH_KILOMETERS,
+    LENGTH_MILES,
+    PERCENTAGE,
+    PRESSURE_PSI,
+)
+
+from . import MazdaEntity
+from .const import DATA_COORDINATOR, DOMAIN
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the sensor platform."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][DATA_COORDINATOR]
+
+    entities = []
+
+    if coordinator.data is not None:
+        for index, _ in enumerate(coordinator.data):
+            entities.append(MazdaFuelRemainingSensor(coordinator, index))
+            entities.append(MazdaFuelDistanceSensor(coordinator, index))
+            entities.append(MazdaOdometerSensor(coordinator, index))
+            entities.append(MazdaFrontLeftTirePressureSensor(coordinator, index))
+            entities.append(MazdaFrontRightTirePressureSensor(coordinator, index))
+            entities.append(MazdaRearLeftTirePressureSensor(coordinator, index))
+            entities.append(MazdaRearRightTirePressureSensor(coordinator, index))
+
+    async_add_entities(entities)
+
+
+class MazdaFuelRemainingSensor(MazdaEntity):
+    """Class for the fuel remaining sensor."""
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        vehicle_name = self.get_vehicle_name()
```

Comment:
Can the vehicle name change dynamically? Or is this always the same? 

If it's always the same, you'll get slightly better performance by calculating the name and the unique ID in the constructor. You could take `name_part` and `unique_id` part in the base class and have each entity pass their unique part in the constructor.

Reply:
Actually, the vehicle name can change dynamically. The user can rename their vehicle in the MyMazda app at any time, which would cause the name returned by the API to be different.